core/state/statedb.go

```diff
@@ -75,7 +75,7 @@ type StateDB struct {
 	snapAccounts  map[common.Hash][]byte
 	snapStorage   map[common.Hash]map[common.Hash][]byte
 
-	mutex        sync.Mutex
+	mutex        sync.RWMutex
```

Comment:
Should we keep `Mutex`? 

- In `Mutex`, we block other read-only threads from accessing the `state` during `Copy()`.
- In `RWMutex`, we allow multiple read-only threads to access the `state` during `Copy()`

My concern with keeping `Mutex` is:
- Given `Copy()` is an expensive function to read, since we are copying the entire `stateDb`, blocking with `mutex.lock()` would cause performance issues, especially when there are multiple routines that want to read from the `state`
- If the quorum nodes are used in a read heavy manner (cache-like), then `RWMutex` is the way to go here